src/server_manager/ui_components/outline-server-settings.html

```diff
@@ -245,72 +247,67 @@ <h3>[[localize('settings-metrics-header')]]</h3>
-      },
-      _setKeysPortButtonsEnabled: function(enabled) {
-        this.$.keysPortSaveButton.disabled = !enabled;
-        this.$.keysPortCancelButton.disabled = !enabled;
-      },
-      _handleKeysPortTap: function() {
+      setKeysPortEditing: function() {
         if (!this.isAccessKeyPortEditable) { 
           return;
         }
-        this._setKeysPortButtonsVisible(true);
+        this.showKeysPortButtons = true;
         const invalid = this.$.keysPortInput.invalid;
         this.$.keysPortSaveButton.disabled = invalid;
         this.$.keysPortCancelButton.disabled = false;
       },
-      _handleKeysPortServerSuccess() {
+      setKeysPortSaved: function() {
         this.serverPortForNewAccessKeys = Number(this.$.keysPortInput.value);
-        this._setKeysPortButtonsVisible(false);
+        this.showKeysPortButtons = false;
         this.$.keysPortInput.readonly = false;
       },
-      _handleKeysPortServerFail(message) {
+      setKeysPortErrorState: function(message) {
         const input = this.$.keysPortInput;
-        this._setKeysPortButtonsEnabled(true); 
+        this.enableKeysPortButtons = true; 
         input.invalid = true;
         input.errorMessage = message;
         input.readonly = false;
         input.focus();
       },
-      _handleKeysPortSave: function() {
+      saveKeysPort: function() {
         const input = this.$.keysPortInput;
         if (Number(input.value) === this.serverPortForNewAccessKeys) {
-          this._handleKeysPortCancel();
+          this.cancelKeysPortEdit();
           return;
         };
-        this._setKeysPortButtonsEnabled(false);
+        this.enableKeysPortButtons = false;
         input.readonly = true;
         input.invalid = false;
         this.fire('ChangePortForNewAccessKeysRequested', {
           port: Number(this.$.keysPortInput.value), 
-          success: this._handleKeysPortServerSuccess.bind(this), 
-          fail: this._handleKeysPortServerFail.bind(this)
+          serverSettings: this
         });
       },
-      _handleKeysPortCancel: function() {
+      cancelKeysPortEdit: function() {
         this.$.keysPortInput.value = this.serverPortForNewAccessKeys;
         this.$.keysPortInput.invalid = false;
-        this._setKeysPortButtonsVisible(false);
+        this.showKeysPortButtons = false;
       },
-      _handleKeysPortInputKeyDown: function(e) {
+      _handleKeysPortInputKeyUp: function(e) {
         let input = this.$.keysPortInput;
+        const badInputMessage = this.localize('error-keys-port-bad-input');
         if (event.key === 'Escape') {
-          this._handleKeysPortCancel();
+          this.cancelKeysPortEdit();
           input.blur();
           return; 
-        } else if (event.key === 'Enter' && !input.invalid) {
-          this._handleKeysPortSave();
+        } else if (event.key === 'Enter') {
+          if (input.invalid && input.errorMessage === badInputMessage) {
```

Comment:
Why do you need to check the error message? Won't the input.invalid give you the information you need?

Reply:
My mistake, fixed the issue